crates/ott-balancer/src/discovery/dns.rs

```diff
@@ -0,0 +1,51 @@
+use async_trait::async_trait;
+use tracing::info;
+use trust_dns_resolver::TokioAsyncResolver;
+
+use super::*;
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct DockerDiscoveryConfig {
+    /// The port that monoliths should be listening on for load balancer connections.
+    pub monolith_port: u16,
+    pub dns_server: Option<String>,
+}
+
+pub struct DockerMonolithDiscoverer {
+    config: DockerDiscoveryConfig,
+    query: Option<String>,
```

Comment:
This needs be configurable. move this to the config above and make it required.

Reply:
So move just the query, right?

Reply:
yes, and make it not `Option`